test/functional/tests/cache_ops/test_dynamic_cache_mode_switching.py

```diff
@@ -0,0 +1,321 @@
+
+    with TestRun.step("Stop all caches"):
+        casadm.stop_all_caches()
+        Udev.enable()
+
+
+@pytest.mark.parametrize(
+    "cache_mode_1,cache_mode_2,flush",
+    [
+        (CacheMode.WT, CacheMode.WB, False),
+        (CacheMode.WB, CacheMode.PT, False),
+        (CacheMode.WB, CacheMode.PT, True),
+        (CacheMode.WB, CacheMode.WT, False),
+        (CacheMode.WB, CacheMode.WT, True),
+        (CacheMode.PT, CacheMode.WT, False),
+        (CacheMode.WT, CacheMode.WA, False),
+    ],
+)
+@pytest.mark.parametrize("io_mode", [ReadWrite.randwrite, ReadWrite.randrw])
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_cache_mode_switching_during_io(cache_mode_1, cache_mode_2, flush, io_mode):
+    """
+        title: Test for dynamic cache mode switching during IO.
+        description: |
+          Validate the ability of the CAS to switch cache modes
+          during working IO on CAS device.
+        pass_criteria:
+          - Cache mode is switched without errors.
+    """
+
+    with TestRun.step("Partition cache and core devices"):
+        cache_dev, core_dev = storage_prepare()
+
+    with TestRun.step(f"Start cache in {cache_mode_1} mode"):
+        cache = casadm.start_cache(cache_dev, cache_mode_1, force=True)
+
+    with TestRun.step("Add core to the cache"):
+        core = cache.add_core(core_dev)
+
+    with TestRun.step("Run 'fio'"):
+        fio = (
+            Fio()
+            .create_command()
+            .io_engine(IoEngine.libaio)
+            .block_size(Size(4, Unit.KibiByte))
+            .read_write(io_mode)
+            .target(core.system_path)
+            .direct(1)
+            .run_time(timedelta(minutes=4))
+            .time_based()
+        )
+        fio_pid = fio.run_in_background()
+        time.sleep(5)
+
+    with TestRun.step(
+        f"Change cache mode to {cache_mode_2} with flush cache set to: {flush}"
+    ):
+        cache_mode_switch_output = cache.set_cache_mode(cache_mode_2, flush)
+
+    with TestRun.step(f"Check if cache mode has switched properly during IO"):
+        cache_mode_after_switch = cache.get_cache_mode()
+
+        if (
+            cache_mode_switch_output.exit_code == 0
+            and cache_mode_after_switch == cache_mode_2
+        ):
+            TestRun.LOGGER.info("Cache mode switched properly")
+        else:
+            TestRun.fail(
+                f"Cache mode did not switch properly! "
+                f"Exit code: {cache_mode_switch_output.exit_code}, "
+                f"Cache mode after switch: {cache_mode_after_switch}"
+            )
+
+    with TestRun.step("Stop 'fio'"):
+        TestRun.executor.kill_process(fio_pid)
+
+    with TestRun.step("Stop all caches"):
+        casadm.stop_all_caches()
+
+
+def storage_prepare():
+    cache_dev = TestRun.disks["cache"]
+    cache_dev.create_partitions([Size(2, Unit.GibiByte)])
```

Comment:
Is it intended that cache is bigger than core?

Reply:
Changed to bigger core than cache.